plugins/azure/resoto_plugin_azure/__init__.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import multiprocessing
+from collections import namedtuple
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import Optional, Tuple, Any
+
+from resoto_plugin_azure.collector import AzureSubscriptionCollector
+from resoto_plugin_azure.config import AzureConfig, AzureAccountConfig
+from resoto_plugin_azure.resource.base import AzureSubscription
+from resotolib.baseplugin import BaseCollectorPlugin
+from resotolib.baseresources import Cloud
+from resotolib.config import Config
+from resotolib.core.actions import CoreFeedback
+from resotolib.core.progress import ProgressTree, ProgressDone
+from resotolib.graph import Graph
+from resotolib.proc import collector_initializer
+
+log = logging.getLogger("resoto.plugin.azure")
+
+AzureSubscriptionArg = namedtuple(
+    "AzureSubscriptionArg", ["config", "cloud", "subscription", "account_config", "core_feedback"]
+)
+
+
+class AzureCollectorPlugin(BaseCollectorPlugin):
+    cloud = "azure"
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.core_feedback: Optional[CoreFeedback] = None
+
+    @staticmethod
+    def add_config(cfg: Config) -> None:
+        cfg.add_config(AzureConfig)
+
+    @staticmethod
+    def auto_enableable() -> bool:
+        return False
+
+    def collect(self) -> None:
+        log.info("plugin: Azure collecting resources")
+        config: AzureConfig = Config.azure
+        assert self.core_feedback, "core_feedback is not set"
+        cloud = Cloud(id=self.cloud, name="Azure")
+
+        # In case no account is configured, fallback to default settings
+        account_configs = config.accounts or [AzureAccountConfig()]
+
+        # Gather all subscriptions
+        args = [
+            AzureSubscriptionArg(
+                config, cloud, subscription, ac, self.core_feedback.with_context(cloud.id, subscription.safe_name)
+            )
+            for ac in account_configs
+            for subscription in AzureSubscription.list_subscriptions(ac.credentials())
+        ]
+
+        # Send initial progress
+        progress = ProgressTree(self.cloud)
+        for sub in args:
+            progress.add_progress(ProgressDone(sub.subscription.subscription_id, 0, 1))
+            log.debug(f"Found {sub.subscription.subscription_id}")
+        self.core_feedback.progress(progress)
+
+        # Collect all subscriptions
+        with ThreadPoolExecutor(max_workers=config.resource_pool_size) as executor:
```

Comment:
This should probably be a dedicated setting like `config.subscription_pool_size`.

Reply:
It is a dedicated setting in the config. I will rename it.